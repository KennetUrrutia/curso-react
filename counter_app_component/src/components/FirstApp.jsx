import PropTypes from 'prop-types'

const FirstApp = (props) => {

    const { title, subTitle, name } = props


  return (
    <>
        <h1 data-testid='test-title'> { title } </h1>
        <p>{ subTitle }</p>
        <p>{ subTitle }</p>
        <p>{ name }</p>
    </>
  )
}


FirstApp.propTypes = {
    title: PropTypes.string.isRequired,
    subTitle: PropTypes.string
}

FirstApp.defaultProps = {
    name: 'Kennet Urrutia',
    subTitle: 'Programador JR',
    // title: 'Ingeniero en sistemas'
}
export default FirstApp