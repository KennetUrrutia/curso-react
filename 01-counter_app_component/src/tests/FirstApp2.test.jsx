import { render, screen } from "@testing-library/react"
import FirstApp from "../components/FirstApp"

describe('Pruebas en <FirstApp />', () => {

  const title = 'Hola soy Kennet'
  const subTitle = 'Hola, soy un subtitulo'

  test('debe de hacer match con el snapshot', () => {
    const { container } = render(
      <FirstApp
        title={title}
        subTitle={subTitle}
      />
    )

    expect(container).toMatchSnapshot() //Crear un snapshot
  })

  test('debe de mostrar el mensaje de "Hola, soy Kennet"', () => {
    render(
      <FirstApp
        title={title}
        subTitle={subTitle}
      />
    )

    expect(screen.getByText(title)).toBeTruthy() // toBeTruthy es para ver si existe

  })

  test('debe de mostrar el titulo en un h1', () => {
    render(
      <FirstApp
        title={title}
        subTitle={subTitle}
      />
    )
    expect(screen.getByRole('heading', { level: 1 }).innerHTML).toContain(title)
  })

  test('debe de mostrar el subtitulo enviado por props', () => {
    render(
      <FirstApp
        title={title}
        subTitle={subTitle}
      />
    )

    expect(screen.getAllByText(subTitle).length).toBe(2)

  })
})