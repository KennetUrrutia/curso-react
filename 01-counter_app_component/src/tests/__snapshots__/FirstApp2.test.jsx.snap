// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <FirstApp /> debe de hacer match con el snapshot 1`] = `
<div>
  <h1
    data-testid="test-title"
  >
     
    Hola soy Kennet
     
  </h1>
  <p>
    Hola, soy un subtitulo
  </p>
  <p>
    Hola, soy un subtitulo
  </p>
  <p>
    Kennet Urrutia
  </p>
</div>
`;
